### ISSUE TITLE
Workspace Review module not showing document nodes created in different dimensions

### ISSUE THREAD

**Issue by franklinkim** [neos/neos-development-collection#1458]
### Description

As discussed with @kdambekalns:
If you create a document node in one dimension and edit it in another, the workspace review module only shows the initial node but not the one in the other dimension. 

Notes:
Unpublished content elements of these two nodes are being showed.
If you just publish the document node in the one dimension it shows it will show the unpublished dimension afterwards.

### Steps to Reproduce

Dimension Config:

```yaml
TYPO3:
  TYPO3CR:
    contentDimensions:
      language:
        default: en
        defaultPreset: en
        icon: icon-language
        label: Language
        presets:
          en:
            label: English
            uriSegment: en
            values:
              - en
          de:
            label: German
            uriSegment: de
            values:
              - de
              - en
      country:
        default: all
        defaultPreset: all
        label: Country
        icon: icon-globe
        presets:
          all:
            label: International
            values:
              - all
            uriSegment: all
          de:
            label: Germany
            values:
              - de
              - all
            uriSegment: de

```

Workspace setup: Shared Workspace "review" based on "live"

1. Create Page *test* in dimension `language: en, country: all`
2. Switch dimension to  `language: en, country: de`
3. Edit the Title
4. Publish all to `review` Workspace

#### Expected behavior

Show unpublished changes for *test* page in the dimensions

* `language: en, country: all`
* `language: en, country: de`

#### Actual behavior

Only shows unpublished changes for *test* page in the dimensions

* `language: en, country: all`

### Affected Versions

Neos: 2.3.10

Flow: 3.3.8

**Comment by jobee** [comment]
Since this is open for quite a while now, is this aggregation/hiding of node changes in other dimensions an expected behaviour or a bug that should be fixed?

I guess, the dimensions should be part of the Array keys?  see...
https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php#L496

**Comment by franklinkim** [comment]
yes, it's been a while... guess the expected behaviour would be to see changes per dimension or on what dimensions the changes occurred. 
Someone might just want to publish changes to one dimension only...

**Comment by kitsunet** [comment]
I just found this as well... and this is not only about document nodes but also content nodes. In general this is pretty messy and I guess it was just not niced because it can be very subtle. Looking into options to fix....

**Comment by Sebobo** [comment]
Duplicate of #3470 and will be fixed via #3986.
